site/src/Consumer/AbstractNotificationConsumer.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Consumer;
+
+use App\Util\Notification;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class AbstractNotificationConsumer
+ *
+ * Base class for all the Notification Consumers that we have
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+abstract class AbstractNotificationConsumer implements ConsumerInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * AbstractNotificationConsumer constructor.
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param AMQPMessage $msg
+     *
+     * @return bool|mixed
+     *
+     * @throws \Exception
+     */
+    public function execute(AMQPMessage $msg)
+    {
+        try {
+            $notification = unserialize($msg->body);
+
+            return $this->handle($notification);
+        } catch (\Exception $e) {
+            throw $e;
+        }
```

Comment:
Don't throw an exception here, it will abort the handler. Please look at the existing execute() methods in other code to see the possible return codes from this function.